app/views/companies/index.html.erb

```diff
@@ -0,0 +1,28 @@
+<div class="container">
+<h3>All Companies</h3>
+<div>
+  <%= link_to "View by location", "/companies?sort=location" %>
+</div>
+<% @companies.each do |company| %>
+  <ul class=company_<%= company.id %>>
+    <%= link_to company.name, company_path(company) %> - <%= company.city %><br>
+    <%= link_to edit_company_path(company) do %>
+    <button type="button" class="btn btn-success" aria-haspopup="true" aria-expanded="false">
+      <span class="glyphicon glyphicon-pencil" aria-hidden="true"></span>
+    </button>
+    <% end %>
+    <%= link_to company_path(company), method: "delete" do %>
+    <button type="button" class="btn btn-danger" aria-haspopup="true" aria-expanded="false"> Delete
+    </button>
+    <% end %>
+  </ul>
+<% end %>
+
+<!-- Single button -->
+<div class="btn-group">
+  <a href="/companies/new"><button type="button" class="btn btn-success" aria-haspopup="true" aria-expanded="false">
```

Comment:
be sure to leverage rails link_to helper and your path helpers from your routes rather than hardcoding your links!